### PR TITLE
Build Chocolatey Package in Travis CI

### PR DESCRIPTION
**Pull request by michaelkschmidt**
This will build a Chocolatey Package as part of the CI build scripts.

~~Note I haven't been able to test with Travis CI yet.  The scripts it calls are all working, but the choco dependencies might need to be tweaked~~  Edit: Looks like Travis CI will run from this PR.

### CODE REVIEW

File: scripts/build_pkg.sh [hunk 1]
```diff
@@ -39,6 +39,7 @@ fi
 DEPS_INSTALL_DIR=$BUILD_DIR/$CROSS_COMPILE/deps/usr
 FWUP_INSTALL_DIR=$BUILD_DIR/$CROSS_COMPILE/fwup-staging/usr
 PKG_CONFIG_PATH=$DEPS_INSTALL_DIR/lib/pkgconfig
+export ChocolateyInstall=$DEPS_INSTALL_DIR/chocolatey
```
Comment (fhunleth) [inline scripts/build_pkg.sh]: Does `ChocolateyInstall` need to be exported? If not, then I'd prefer it to be `CHOCOLATEY_DIR` to be more consistent with the other variables.

Comment (michaelkschmidt) [inline scripts/build_pkg.sh]: Unfortunately, choco won't run without that set.   Want to me to move it line 120 to make that clearer/cleaner?

Comment (fhunleth) [inline scripts/build_pkg.sh]: That sounds good. Also, while you're at it, could you rebase your intermediate commits together? It looks like there should just be one commit in the end for this change, but if there's an unrelated change in there that I missed, please separate that out.

File: scripts/fwup.nuspec [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>fwup</id>
+    <title>fwup</title>
+    <version>0.9.1</version>
```
Comment (GregMefford) [inline scripts/fwup.nuspec]: Do you agree that it would be better to dynamically populate this version from [the VERSION file](https://github.com/fhunleth/fwup/blob/master/VERSION) at build time so we wouldn't have to remember to update it when packaging a new build?

For example, you could put `%VERSION%` in the `<version>` tag in the checked-in `fwup.nuspec` file, and have a step in the `build_pkg.sh` script, just before the `choco.exe pack` command, that would run `sed -i "s/%VERSION%/$(cat VERSION)/" fwup.nuspec` to finalize the metadata before using it.

File: scripts/fwup.nuspec [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <authors>Frank Hunleth</authors>
+    <owners>Greg Mefford,Michael Schmidt</owners>
+    <summary>Configurable embedded Linux firmware update creator and runner</summary>
+    <description>The fwup utility is a configurable image-based firmware update utility for embedded Linux-based systems. It has two modes of operation. The first mode creates compressed archives containing root file system images, bootloaders, and other image material. These can be distributed via websites, email or update servers. The second mode applies the firmware images in a robust and repeatable way.</description>
+    <releaseNotes>https://github.com/fhunleth/fwup/releases</releaseNotes>
+    <projectUrl>https://github.com/fhunleth/fwup</projectUrl>
+    <tags></tags>
+    <packageSourceUrl>https://github.com/fhunleth/fwup</packageSourceUrl>
+    <docsUrl>https://github.com/fhunleth/fwup</docsUrl>
```
Comment (GregMefford) [inline scripts/fwup.nuspec]: I don't think it makes a big difference, but I think the `docsUrl` (as opposed to the `projectSourceUrl`) should be https://github.com/fhunleth/fwup/blob/master/README.md

File: scripts/fwup.nuspec [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <bugTrackerUrl>https://github.com/fhunleth/fwup/issues</bugTrackerUrl>
+    <projectSourceUrl>https://github.com/fhunleth/fwup</projectSourceUrl>
+    <licenseUrl>https://github.com/fhunleth/fwup/blob/master/LICENSE</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <iconUrl>https://cdn.rawgit.com/fhunleth/fwup/master/docs/fwup-pup256x256.png</iconUrl>
```
Comment (GregMefford) [inline scripts/fwup.nuspec]: Per the [Chocolatey package icon guidelines](https://github.com/chocolatey/choco/wiki/CreatePackages#package-icon-guidelines), it's best to point this to an immutable tag instead of a dynamic reference like `master`, otherwise `cdn.rawgit.com` will cache the current master forever, even if it changes later.

I think if you change it to `https://cdn.rawgit.com/fhunleth/fwup/v%VERSION%/docs/fwup-pup256x256.png` (note the `v` before `%VERSION%`) and do the `sed` script thing above, it should do what we want.

**Comment by GregMefford** [comment]
Really great work, @michaelkschmidt! I just had a few comments inline above to polish it off. 🚀
